File: AddressBook/AddressBook/Dtos/ContactChat.cs
using AddressBook.Entities;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using AddressBook.Common.Paging;

namespace AddressBook.Dtos
{
    public class ContactChatRes
    {
        public int ContactChatId { get; set; }
        public string? Chat { get; set; }

        public int? ContactId { get; set; }
        public ContactRes? Contact { get; set; }

        public int? ChatLabelId { get; set; }
        public ChatLabelRes? ChatLabel { get; set; }
    }

    public class ContactChatReqEdit
    {
        [Required]
        public string? Chat { get; set; }
        public int? ContactId { get; set; }
        public int? ChatLabelId { get; set; }
    }

    public class ContactChatReqSearch : PagedReq
    {
        public int ContactId { get; set; }
    }
}

File: AddressBook/react-client/src/dtos/ContactChat.ts
import Common from "../utility/Common";
import { ChatLabelRes } from "./ChatLabel";
import { PagedReq } from "./PagedReq";
import { ContactRes } from "./Contact";

export interface ContactChatRes {
  contactChatId?: string;
  chat?: string;
  contactId?: string;
  contact?: ContactRes;
  chatLabelId?: string;
  chatLabel?: ChatLabelRes;
}

export class ContactChatReqEdit {
  chat?: string = "";
  contactId?: string = "";
  chatLabelId?: string = "";
  constructor(contactId?: string) {
    this.contactId = contactId;
  }
}

export class ContactChatReqSearch extends PagedReq {
  contactId?: string;
  constructor(
    {
      pageNumber = 1,
      pageSize = Common.DEFAULT_PAGE_SIZE,
      orderBy,
      searchText = "",
    }: PagedReq,
    {contactId = ""}
  ) {
    super({
      pageNumber: pageNumber,
      pageSize: pageSize,
      orderBy: orderBy,
      searchText: searchText,
    });
    this.contactId = contactId;
  }
}